src/crypto/public_key/elliptic/ecdh_x.js

```diff
@@ -24,12 +25,27 @@ const HKDF_INFO = {
  */
 export async function generate(algo) {
   switch (algo) {
-    case enums.publicKey.x25519: {
-      // k stays in little-endian, unlike legacy ECDH over curve25519
-      const k = getRandomBytes(32);
-      const { publicKey: A } = x25519.box.keyPair.fromSecretKey(k);
-      return { A, k };
-    }
+    case enums.publicKey.x25519:
+      try {
+        const webCrypto = util.getWebCrypto();
+        // Note: keys generated with ECDSA and ECDH are structurally equivalent
+        const webCryptoKey = await webCrypto.generateKey({ name: 'X25519' }, true, ['deriveKey', 'deriveBits']);
+
+        const privateKey = await webCrypto.exportKey('jwk', webCryptoKey.privateKey);
+        const publicKey = await webCrypto.exportKey('jwk', webCryptoKey.publicKey);
+
+        return {
+          A: new Uint8Array(b64ToUint8Array(publicKey.x)),
+          k: b64ToUint8Array(privateKey.d, true)
+        };
+      } catch (err) {
+        // X25519 not implemented by WebCrypto
+        // k stays in little-endian, unlike legacy ECDH over curve25519
+        const k = getRandomBytes(32);
+        const { publicKey: A } = x25519.box.keyPair.fromSecretKey(k);
+        return { A, k };
```

Comment:
Also here I think it's clearer and potentially safer to include the conditional:
```suggestion
        if (err.name === 'NotSupportedError') {
          // k stays in little-endian, unlike legacy ECDH over curve25519
          const k = getRandomBytes(32);
          const { publicKey: A } = x25519.box.keyPair.fromSecretKey(k);
          return { A, k };
        }
        throw err;
```
(theoretically it could also throw other errors, like being out of memory, although ofc that's very unlikely)